hare/hare_shared.go

```diff
@@ -0,0 +1,43 @@
+package hare
```

Comment:
why did you call it hare_shared? what is the purpose of splitting this code from the hare protocol?